language-plutus-core/generators/Language/PlutusCore/Generators/Internal/TypedBuiltinGen.hs

```diff
@@ -144,87 +125,15 @@ genTypedBuiltinFail tb = fail $ fold
 genTypedBuiltinDef :: Monad m => TypedBuiltinGenT m
 genTypedBuiltinDef
     = updateTypedBuiltinGenInt
-          (\low high -> Gen.integral $ Range.linearFrom 0 low high)
+         (Gen.integral $ Range.linearFrom 0 0 10)
```

Comment:
Have you tried making the upper bound `2` rather than `10`? It may or may not help with the looping.

Reply:
Yeah I tried that and it didn't help...